### PR TITLE
Generate random beta_access_code if none present

### PR DESCRIPTION
**Pull request by robbiecarlton**

### CODE REVIEW

**Comment by robbiecarlton** [comment]
@levity - good to merge?

File: api/models/Community.js [hunk 1]
```diff
@@ -164,5 +165,19 @@ module.exports = bookshelf.Model.extend({
       var slackMessage = Slack.textForNewPost(post, community)
       return Slack.send(slackMessage, community.get('slack_hook_url'))
     })
+  },
+
+  getNewBetaAccessCode: function () {
+    const test = code => Community.where({beta_access_code: code}).fetch()
```
Comment (levity) [inline api/models/Community.js]: `Community.where({beta_access_code: code}).count()` works too

File: api/models/Community.js [hunk 2]
```diff
@@ -164,5 +165,19 @@ module.exports = bookshelf.Model.extend({
...
+    const loop = code =>
+      test(code)
+      .then(community => {
+        if (community) {
+          return loop(randomstring.generate({length: 6, charset: 'alphanumeric'}))
+        } else {
+          return code
+        }
+      })
+    return loop(randomstring.generate({length: 6, charset: 'alphanumeric'}))
```
Comment (levity) [inline api/models/Community.js]: you could avoid repeating the `generate` line by just doing it in the beginning of `loop`

Comment (levity) [inline api/models/Community.js]: makes the whole thing more concise too:
```js
const test => code => Community.where({beta_access_code: code}).count().then(Number)
const loop = () => {
  const code = randomstring.generate({length: 6, charset: 'alphanumeric'})
  return test(code).then(count => count ? loop() : code)
}
return loop()
```

**Comment by levity** [comment]
feel free to merge after making the changes above

**Pull request merged**